### PR TITLE
Remove some JS jasmine snippets

### PR DESCRIPTION
**Pull request by lpil**
There's quite a lot of removal here. What do you think?

https://github.com/honza/vim-snippets/issues/468#issuecomment-102473202

### CODE REVIEW

File: UltiSnips/javascript_jasmine.snippets [hunk 1]
```diff
@@ -1,10 +1,6 @@
 priority -50
 
-# JavaScript versions -- from the TextMate bundle + some additions
-# for jasmine-jquery matchers
-#
-
-snippet des "Describe (js)" b
+snippet desc "Describe (js)" b
```
Comment (SirVer) [inline UltiSnips/javascript_jasmine.snippets]: use `describe` as trigger

File: UltiSnips/javascript_jasmine.snippets [hunk 2]
```diff
@@ -16,169 +12,58 @@ it('${1:description}', function() {
 });
 endsnippet
 
-snippet bef "before each (js)" b
```
Comment (SirVer) [inline UltiSnips/javascript_jasmine.snippets]: this one is very useful. `before` is an acceptable trigger though

File: UltiSnips/javascript_jasmine.snippets [hunk 3]
```diff
@@ -16,169 +12,58 @@ it('${1:description}', function() {
...
-beforeEach(function() {
-	$0
-});
-endsnippet
-
-snippet aft "after each (js)" b
```
Comment (SirVer) [inline UltiSnips/javascript_jasmine.snippets]: same here

**Comment by SirVer** [comment]
a lot of these snippets are very useful to me on a daily basis. How about we just pull them out of javascript into another filetype?

File: UltiSnips/javascript_jasmine.snippets [hunk 4]
```diff
@@ -16,169 +12,58 @@ it('${1:description}', function() {
...
-afterEach(function() {
-	$0
-});
-endsnippet
-
-snippet any "any (js)" b
-jasmine.any($1)
-endsnippet
-
-snippet ru "runs (js)" b
```
Comment (SirVer) [inline UltiSnips/javascript_jasmine.snippets]: `runs` is a better trigger, but that is useful.

File: UltiSnips/javascript_jasmine.snippets [hunk 5]
```diff
@@ -16,169 +12,58 @@ it('${1:description}', function() {
...
-runs(function() {
-	$0
-});
-endsnippet
-
-snippet wa "waits (js)" b
```
Comment (SirVer) [inline UltiSnips/javascript_jasmine.snippets]: same here, but 'waits'

File: UltiSnips/javascript_jasmine.snippets [hunk 6]
```diff
@@ -16,169 +12,58 @@ it('${1:description}', function() {
...
-waits($1);
-endsnippet
-
-snippet ex "expect (js)" b
```
Comment (SirVer) [inline UltiSnips/javascript_jasmine.snippets]: they are used very often in jasmine tests - I approve that the triggers are short. `ee` makes more sense for expect to equal than expe, so I do not agree your suggested change is ergonomical.

File: UltiSnips/javascript_jasmine.snippets [hunk 7]
```diff
@@ -16,169 +12,58 @@ it('${1:description}', function() {
...
-snippet en "expect to be null (js)" b
-expect(${1:target}).toBeNull();
-endsnippet
-
-snippet ec "expect to contain (js)" b
+snippet expc "expect to contain (js)" b
 expect(${1:target}).toContain(${2:value});
 endsnippet
 
-snippet ev "expect to be visible (js)" b
```
Comment (SirVer) [inline UltiSnips/javascript_jasmine.snippets]: I think it is fine to remove these - they are rather obscure.

**Comment by SirVer** [comment]
mmh.... those snippets *are* super useful. I hate to see them go. Also, short triggers for common matchers seem fine to me. Why do you want to cut them down?

File: UltiSnips/javascript_jasmine.snippets (hunk 1 above)
Comment (lpil) [inline UltiSnips/javascript_jasmine.snippets]: `desc` is the trigger used in other popular BDD frameworks, I think it's probably better to keep things consistent, as recommended in the readme.

File: UltiSnips/javascript_jasmine.snippets (hunk 6 above)
Comment (lpil) [inline UltiSnips/javascript_jasmine.snippets]: I disgree `ee` is far too short a trigger for this.

Other BDD testing framework snippets I've looked at don't have many snippets like these, and the few that do have triggers like the one used here, or longer.

**Comment by lpil** [comment]
Because the Javascript namespace is a mess of library specific snippets and these often clash.

Additionally they don't match many of the other BDD test framework triggers.

If we want to keep more of them I say we move them (and the other JS library snippets) out of the default JS namespace

**Comment by SirVer** [comment]
> If we want to keep more of them I say we move them (and the other JS library snippets) out of the default JS namespace

I think that is a good approach to take. Let's teach people how to include the ones they want, i.e. using `:UltiSnipsAddFiletypes`.

**Comment by mmwtsn** [comment]
+1 for moving these out of the JavaScript namespace and letting people add them in as needed.

If we clear up the JavaScript namespace we'll be able to support more testing libraries—Qunit, Mocha, Chai's various assertion styles, etc.—using sane triggers (e.g. `before`) without collision.

**Pull request closed without merging**